serf/serf.go

```diff
@@ -1577,6 +1583,10 @@ func (s *Serf) reap(old []*memberState, now time.Time, timeout time.Duration) []
 	for i := 0; i < n; i++ {
 		m := old[i]
 
+		if s.config.ReconnectTimeoutOverride != nil {
+			timeout = s.config.ReconnectTimeoutOverride.ReconnectTimeout(&m.Member, timeout)
```

Comment:
This loops over several members, but the original `timeout` from the args is mutated here on _each_ loop, rather than doing something like `memberTimeout := timeout` just before this conditional, and then passing `memberTimeout` into the `ReconnectTimeout` inputs and assigning to the output.

Reply:
Good catch